### PR TITLE
First files for 1.0.0.216 #283

### PR DESCRIPTION
**Pull request by oerms**
First set of files for R&F patch.

### CODE REVIEW

**Comment by oerms** [comment]
Probably won't be able to work on much else until tomorrow or Monday. So please merge, if you like.

**Comment by oerms** [comment]
I'll be committing to this branch if I get find the time. I just want to make sure, it is somewhere visible, what I am working on to avoid duplicate work...

File: Assets/UI/civ6common.lua [hunk 1]
```diff
@@ -112,12 +112,18 @@ function MoveUnitToPlot( kUnit:table, plotX:number, plotY:number )
...
+      end
+    end
+ 
     if (bWillStartWar) then
-      local eDefendingPlayer = CombatManager.GetBestDefender(eUnitComponentID, plotX, plotY );
-      if (eDefendingPlayer == nil) then
-        local pPlot = Map.GetPlot(plotX, plotY);
-        eDefendingPlayer = pPlot:GetOwner();
+      local eDefendingPlayer = results[1];
       end
```
Comment (Azurency) [inline Assets/UI/civ6common.lua]: this `end` should not be here

**Comment by oerms** [comment]
Does anybody know which Integrations are already being worked on? Any word from the owners?

**Comment by ihendriks** [comment]
@oerms Aside of the checklist on #283 there's only the discord but it's not really used much it seems.

File: Assets/UI/actionpanel.lua [hunk 2]
```diff
@@ -595,6 +594,8 @@ function DoEndTurn( optionalNewBlocker:number )
       if(attackCity ~= nil) then
           LuaEvents.CQUI_Strike_Enter();
           LuaEvents.CQUI_CityRangeStrike(Game.GetLocalPlayer(), attackCity:GetID());
+          -- oerms: The following line has been introduced by vanilla Rise and Fall
```
Comment (oerms) [inline Assets/UI/actionpanel.lua]: @Azurency 
Please have a look, if we need to call this function here.

File: Assets/UI/Panels/unitpanel.lua [hunk 3]
```diff
@@ -918,7 +922,8 @@ function View(data)
   end
 
   -- Populate Earned Promotions UI
-  if (not UILens.IsLensActive("Religion") and data.Combat > 0 and (UI.GetInterfaceMode() ~= InterfaceModeTypes.CITY_RANGE_ATTACK) and (UI.GetInterfaceMode() ~= InterfaceModeTypes.DISTRICT_RANGE_ATTACK)) then
+  -- QCUI modified to not allow promotion when in city attack interface (?)
```
Comment (oerms) [inline Assets/UI/Panels/unitpanel.lua]: @Azurency This condition seems to have been modified by CQUI to exclude certain InterfaceModTypes. Please decide whether to keep this way or not.

File: Assets/UI/partialscreenhooks.lua [hunk 4]
```diff
@@ -308,6 +311,9 @@ function OnTutorialCloseAll()
   if IsPartialScreenOpen("WorldRankings") then    LuaEvents.PartialScreenHooks_CloseWorldRankings(); end
 end
 
+-- ===========================================================================
+-- CQUI - function needed for CQUI
+-- ===========================================================================
 function OnAddScreenHook(hookInfo:table)
```
Comment (oerms) [inline Assets/UI/partialscreenhooks.lua]: @Azurency This function is not in the original (pre-patch) game files anymore and seems to be called by CQUI only. Please decide if we still need it; there have been some hooks added with the patch...

**Comment by oerms** [comment]
@Azurency I have added some comments in the diffs...

File: Assets/UI/Civilopedia/civilopediasupport.lua [hunk 5]
```diff
@@ -560,12 +600,24 @@ end
 -- patterns.
 -------------------------------------------------------------------------------
 function FindChapterTextKey(SectionId, PageId, ChapterId, Tag)
-  if(SectionId and PageId and ChapterId and Tag) then
```
Comment (Azurency) [inline Assets/UI/Civilopedia/civilopediasupport.lua]: I think you removed this `if` by accident, it should still be here.

File: Assets/UI/partialscreenhooks.lua (hunk 4 above)
Comment (Azurency) [reply Assets/UI/partialscreenhooks.lua]: It was never in any base game file, it's the "Modular Screen" mod by @astog https://forums.civfanatics.com/threads/modular-screens.620112/ That was integrated, maybe you can update the comment to reflect that it's "Modular Screen".

File: Assets/UI/partialscreenhooks.lua [hunk 6]
```diff
@@ -233,7 +233,12 @@ function OnTurnBegin()
 
   CheckTradeCapacity(localPlayer);
   CheckSpyCapacity(localPlayer);
-
```
[the remark this answers is not in the document]
Comment (oerms) [reply Assets/UI/partialscreenhooks.lua]: I don't get why this does not disappear. I have added the line...

[the remark this answers is not in the document]
Comment (Azurency) [reply Assets/UI/partialscreenhooks.lua]: It's here now 👍

**Comment by Azurency** [comment]
Thank you for your help @oerms 👍. Greatly appreciate it !

**Comment by oerms** [comment]
Any time!

**Pull request merged**